File: uri/media_tres/media_tres.pas
program foo(Input, Output);

uses
    SysUtils;

function media( notaA: Double; notaB: Double; notaC: Double) : Double;
var
    pesoA : Double;
    pesoB : Double;
    pesoC : Double;

begin
    pesoA := 2.0;
    pesoB := 3.0;
    pesoC := 5.0;
    media := ((notaA * pesoA) + (notaB * pesoB) + (notaC * pesoC)) / (pesoA + pesoB + pesoC);
end;

var 
    notaA : Double;
    notaB : Double;
    notaC : Double;
    fullmedia : Double;    

begin
    readLn(notaA);
    readLn(notaB);
    readLn(notaC);
    fullmedia := media(notaA, notaB, notaC);    
    writeLn('MEDIA = ', Format('%.1f', [fullmedia]) );
end.

File: uri/media_tres/media_tres.lua

pesoA = 2.0
pesoB = 3.0
pesoC = 5.0

function media( notaA , notaB, notaC )
    return ((notaA * pesoA) + (notaB * pesoB) + (notaC * pesoC)) / (pesoA + pesoB + pesoC);
end

local notaA, notaB, notaC,  res
notaA = io.read("*n")
notaB = io.read("*n")
notaC = io.read("*n")
res = string.format("%.1f", media( notaA, notaB, notaC ))
print("MEDIA = "..res)
